en/docs/administer/multitenancy/configuring-the-tenant-loading-policy.md

```diff
@@ -49,6 +50,32 @@ If you have Lazy loading enabled, you can configure the allowed tenant idle time
         ```
 
     3.  Restart the server.
+    
+#### Improved artifacts synchronization for Lazy loading
+
+Starting with API Manager version 3.2.0, an in-memory artifacts synchronization mechanism has been introduced, replacing the file-based artifacts management used in earlier versions. This new approach leverages a database to manage artifacts more efficiently.
+
+Following are the key changes introduced by the above mechanism : 
+
+1. **Artifact Storage**
+        
+    Artifacts are now stored in memory and synchronized via a database, moving away from the file-based system.</br>
+
+2. **Dynamic Tenant Unloading**
+     
+    When a tenant is unloaded, its API artifacts are also removed from memory. As a result, if a request is received during the tenant's unloading process, the initial request may fail until the tenant is fully loaded. Once loaded, the gateway can serve the APIs for that tenant seamlessly. 
+    </br></br>Recommended solution for this is to address this limitation and ensure smooth API requests during tenant unloading or loading, you should enable on-demand loading. Add the following configuration to your deployment.toml file:
+        
+        [apim.sync_runtime_artifacts.gateway]
+        gateway_labels = ["Default"]
+        enable_on_demand_loading = true
+    
+    For example, in a multi-tenancy scenario with lazy loading enabled, a user attempting to invoke a tenant's API for the first time after a server restart might encounter a 404 error. This occurs because the corresponding tenant data has not yet been loaded.
```

Comment:
shall we add a description of the error along with the response code

Reply:
addressed the comment and updated the PR